modules/installation-special-config-butane.adoc

```diff
@@ -0,0 +1,105 @@
+
+[TIP]
+====
+Butane releases are backwards-compatible with older releases and with the Fedora CoreOS Config Transpiler (FCCT), unless otherwise noted.
+====
+
+.Procedure
+
+. Navigate to the Butane image download page at https://mirror.openshift.com/pub/openshift-v4/clients/butane/.
+. Get the `butane` binary:
+.. For the newest version of Butane, get the latest `butane` image:
++
+[source,terminal]
+----
+$ curl https://mirror.openshift.com/pub/openshift-v4/clients/butane/latest/butane
+----
++
+.. Optional: For a specific type of architecture you are installing on, such as `butane-aarch64` or `butane-ppc641e`, indicate the appropriate URL. For example:
++
+[source,terminal]
+----
+$ curl https://mirror.openshift.com/pub/openshift-v4/clients/butane/latest/butane-aarch64
+----
++
+. Unpack and unzip the archive.
+. Move the `butane` binary to a directory on your `PATH`.
++
+To check your `PATH`, open a terminal and execute the following command:
++
+[source,terminal]
+----
+$ echo $PATH
+----
+
+After you install the Butane tool, it is available by using the `butane` command:
+
+[source,terminal]
+----
+$ butane <command>
+----
+
+[id="rhcos-creating-butane_{context}"]
+== Creating an Ignition config with Butane
+
+To create machines, you can create a Butane config file that Butane consumes and transpiles into a machine-readable Ignition config file for producing machine configs.
+
+.Prerequisites
+
+* You have added your local SSH private key to the `ssh-agent`. See _Generating an SSH private key and adding it to the agent_ in the "Installing" section.
+* You have installed the `butane` utility.
+
+.Procedure
+
+. Create a Butane config file, such as `99-simple-example.bu`, as shown in the following example:
++
+[source,yaml]
+----
+variant: openshift
+version: 4.8.0
+metadata:
+  name: 99-simple-example
+passwd:
+  users:
+    - name: core
+      ssh_authorized_keys:
+        - ssh-rsa AAAAB3NzaC1yc... <1>
+----
+<1> Set the `ssh_authorized_keys` value to be your SSH public key, which is usually the contents of the `~/.ssh/id_rsa.pub` file on your system.
+
+. Set your ssh public key. On Linux systems, this is usually the contents of the `~/.ssh/id_rsa.pub` file.
+. Create an Ignition config by giving Butane the file that you created in the previous step, for example:
++
+[source,terminal]
+----
+$ ./bin/amd64/butane butane-example.bu
+----
++
+.Example output
+[source,terminal]
+----
+{"ignition":{"config":{"replace":{"source":null,"verification":{}}},"security":{"tls":{}},"timeouts":{},"version":"3.0.0"},"passwd":{"users":[{"name":"core","sshAuthorizedKeys":["ssh-rsa ssh-rsa AAAAB3NzaC1yc..."]}]},"storage":{},"systemd":{}}
+----
++
+An Ignition config is created. You can now provide the Ignition config to a booting {op-system} machine for Ignition to finish configuring your machines.
```

Comment:
If we include this example, should we add the step to generate the YAML file (`butane 99-simple-example.bu --files-dir . -o 99-simple-example.yaml`)?

Reply:
With the `openshift` `4.8.0` spec, Butane generates MachineConfigs directly, and we should update the doc throughout to reflect this.

`--files-dir` is only relevant if `storage.trees` or `storage.files[].contents.local` is specified in the config, and should be omitted otherwise for clarity.